frontend/javascripts/oxalis/view/right-border-tabs/tree_hierarchy_view.tsx

```diff
@@ -688,104 +707,125 @@ function TreeHierarchyView(props: Props) {
-            titleRender={(node) =>
-              node.type === GroupTypeEnum.TREE ? renderTreeNode(node) : renderGroupNode(node)
-            }
-            switcherIcon={<DownOutlined />}
-            onSelect={(_selectedKeys, info: { node: TreeNode; nativeEvent: MouseEvent }) =>
-              info.node.type === GroupTypeEnum.TREE
-                ? onSelectTreeNode(info.node, info.nativeEvent)
-                : onSelectGroupNode(info.node)
-            }
-            onDrop={onDrop}
-            onCheck={onCheck}
-            onExpand={onExpand}
-            // @ts-expect-error isNodeDraggable has argument of base type DataNode but we use it's extended parent type TreeNode
-            draggable={{ nodeDraggable: isNodeDraggable, icon: false }}
-            checkedKeys={checkedKeys}
-            expandedKeys={expandedNodeKeys}
-            selectedKeys={selectedKeys}
-            style={{ marginLeft: -14 }}
-            autoExpandParent
-            checkable
-            blockNode
-            showLine
-            multiple
-            defaultExpandAll
-          />
-        </div>
-      )}
-    </AutoSizer>
+    <>
+      <ContextMenuContainer
+        hideContextMenu={hideContextMenu}
+        contextMenuPosition={contextMenuPosition}
+        menu={menu}
+      />
+      <AutoSizer>
+        {({ height, width }) => (
+          <div
+            style={{
+              height,
+              width,
+            }}
+          >
+            <AntdTree
+              treeData={UITreeData}
+              height={height}
+              ref={treeRef}
+              titleRender={(node) =>
+                node.type === GroupTypeEnum.TREE ? renderTreeNode(node) : renderGroupNode(node)
+              }
+              switcherIcon={<DownOutlined />}
+              onSelect={(_selectedKeys, info: { node: TreeNode; nativeEvent: MouseEvent }) =>
+                info.node.type === GroupTypeEnum.TREE
+                  ? onSelectTreeNode(info.node, info.nativeEvent)
+                  : onSelectGroupNode(info.node)
+              }
+              onDrop={onDrop}
+              onCheck={onCheck}
+              onExpand={onExpand}
+              // @ts-expect-error isNodeDraggable has argument of base type DataNode but we use it's extended parent type TreeNode
+              draggable={{ nodeDraggable: isNodeDraggable, icon: false }}
+              checkedKeys={checkedKeys}
+              expandedKeys={expandedNodeKeys}
+              selectedKeys={selectedKeys}
+              style={{ marginLeft: -14 }}
+              autoExpandParent
+              checkable
+              blockNode
+              showLine
+              multiple
+              defaultExpandAll
+            />
+          </div>
+        )}
+      </AutoSizer>
+    </>
   );
 
-  function setActiveTree(treeId: number) {
-    dispatch(setActiveTreeAction(treeId));
+  function deleteTree(treeId: number) {
+    props.deselectAllTrees();
+    dispatch(deleteTreeAction(treeId));
   }
```

Comment:
For consistency maybe also move this out of the component? If it is necessary to deselectAllTrees just pass it as a param?